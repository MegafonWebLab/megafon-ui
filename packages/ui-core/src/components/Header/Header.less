@import '~styles/base.less';

.mfui-header {
    font-weight: 400;

    &__addition {
        display: inline-block;
        vertical-align: middle;
        margin-top: -2px;
        margin-left: 12px;
    }

    &_level_h1 {
        font-size: 30px;
        line-height: 36px;
        font-weight: 700;

        @media @mobileB {
            font-size:36px;
            line-height: 40px;
        }
        @media @desktop {
            font-size: 44px;
            line-height: 48px;
        }
        @media @desktopMU {
            font-size: 52px;
            line-height: 56px;
        }
    }
    &_margin&_level_h1 {
        margin: 24px 0 32px 0;
    }

    &_level_h2 {
        font-size: 22px;
        line-height: 28px;
        font-weight: 600;
        letter-spacing: 0.5px;

        @media @mobileB {
            font-size: 26px;
            line-height: 32px;
        }
        @media @desktop {
            font-size: 28px;
            line-height: 36px;
        }
        @media @desktopMU {
            font-size: 32px;
            line-height: 40px;
        }
    }
    &_margin&_level_h2 {
        margin: 56px 0 24px 0;
    }

    &_level_h3 {
        font-size: 18px;
        line-height: 24px;
        font-weight: 500;
        letter-spacing: 0.5px;
        @media @desktopMU {
            font-size: 20px;
            line-height: 28px;
        }
    }
    &_margin&_level_h3 {
        margin: 32px 0 24px 0;
    }

    &_level_h5 {
        font-size: 15px;
        line-height: 24px;
        font-weight: 500;
    }
    &_margin&_level_h5 {
        margin: 24px 0 16px 0;
    }

    &_level_h6 {
        font-size: 10px;
        line-height: 14px;
        font-weight: 300;
        @media @desktopMU {
            font-size: 12px;
        }
    }
    &_margin&_level_h6 {
        margin: 0;
    }

    &_color_black {
        color: @soggyAsphalt;
    }
    &_color_green {
        color: @green;
    }
    &_color_purple {
        color: @purple;
    }
    &_color_white {
        color: @clearWhite;
    }
    &_color_blue {
        color: @blue;
    }
    &_color_inherit {
        color: inherit;
    }
}


