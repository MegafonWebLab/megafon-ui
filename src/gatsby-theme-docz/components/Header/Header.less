.docz-header {
    &__h1 {
        margin: 32px 0 24px 0;
    }

    &__h2 {
        margin: 32px 0 16px 0;
    }

    &__h3 {
        margin: 16px 0 8px 0;
    }
}
