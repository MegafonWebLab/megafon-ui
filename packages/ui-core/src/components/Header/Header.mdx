---
name: Header
menu: Components
route: components/header
---

import { Playground, Props } from 'docz';
import Header from './Header';

# Header

## Properties

<Props of={Header} />

## Basic usage

<Playground>
    <Header color="green">Megafon lib</Header>
</Playground>

## Level h2, purple color

<Playground>
    <Header color="purple" as="h2">
        Megafon lib
    </Header>
</Playground>

## Level h3 with margin

<Playground>
    <Header as="h3" margin>
        Megafon lib
    </Header>
</Playground>
