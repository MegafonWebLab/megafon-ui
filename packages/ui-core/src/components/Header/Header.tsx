import * as React from 'react';
import * as PropTypes from 'prop-types';
import './Header.less';
import { cnCreate } from '../../utils/cn';

interface IHeaderProps {
    /** Tag */
    as?: 'h1' | 'h2' | 'h3' | 'h5' | 'h6';
    /** Color */
    color?: 'black' | 'white' | 'green' | 'purple' | 'blue' | 'inherit';
    /** Turn on margin */
    margin?: boolean;
    /** Additional element */
    addition?: JSX.Element;
    /** Custom class name */
    className?: string;
    children?: JSX.Element[] | Element[] | JSX.Element | string | Element;
    /** Click handler */
    onClick?(e: React.SyntheticEvent<EventTarget>): void;
}

const cn = cnCreate('mfui-header');
class Header extends React.Component<IHeaderProps, {}> {
    static propTypes = {
        as: PropTypes.oneOf(['h1', 'h2', 'h3', 'h5', 'h6']),
        color: PropTypes.oneOf(['black', 'white', 'green', 'purple', 'blue', 'inherit']),
        margin: PropTypes.bool,
        addition: PropTypes.element,
        onClick: PropTypes.func,
        children: PropTypes.node,
    };

    static defaultProps: Partial<IHeaderProps> = {
        as: 'h1',
        color: 'black',
    };

    renderAddition() {
        return (
            <div className={cn('addition')}>
                {this.props.addition}
            </div>
        );
    }

    render() {
        const ElementType = this.props.as as string;
        const { color, margin, as: level, onClick } = this.props;

        return (
            <ElementType className={cn('', { color, margin, level }, this.props.className)} onClick={onClick}>
                {this.props.children}
                {this.props.addition && this.renderAddition()}
            </ElementType>
        );
    }
}

export default Header;
