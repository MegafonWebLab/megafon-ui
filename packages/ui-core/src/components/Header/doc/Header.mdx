---
name: Header
menu: Компоненты
route: components/header
---

import { Playground, Props } from 'docz';
import Header from '../Header';
import Example from './Header.example.mdx'

# Header

<CopyToClipboardBox text="import { Header } from '@megafon/ui-core';" />

<DoczTabs>
    <Example />
    <Props of={Header} />
</DoczTabs>
