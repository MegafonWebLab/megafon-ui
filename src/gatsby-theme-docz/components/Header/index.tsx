import React from 'react';
import { cnCreate, Header } from '@megafon/ui-core';
import './Header.less';

const cn = cnCreate('docz-header');

export const h1 = ({ children }) => <Header as='h1' className={cn('h1')} children={children} />;
export const h2 = ({ children }) => <Header as='h2' className={cn('h2')} children={children} />;
export const h3 = ({ children }) => <Header as='h3' className={cn('h3')} children={children} />;
export const h5 = ({ children }) => <Header as='h5' className={cn('h5')} children={children} />;
