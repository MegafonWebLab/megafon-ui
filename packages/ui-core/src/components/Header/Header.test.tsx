import * as React from 'react';
import { shallow } from 'enzyme';
import Header from './Header';

describe('<Header />', () => {
    it('it renders with default h1 tag', () => {
        const wrapper = shallow(<Header />);
        expect(wrapper.type()).toEqual('h1');
    });

    ['h1', 'h2', 'h3', 'h5', 'h6'].forEach(tag => {
        it(`it renders only with ${tag}`, () => {
            const wrapper = shallow(<Header as={tag} />);
            expect(wrapper.type()).toEqual(tag);
        });
    });

    it('it renders with white color class', () => {
        const wrapper = shallow(<Header color="white" />);
        expect(wrapper).toMatchSnapshot();
    });

    it('it renders without class margin', () => {
        const wrapper = shallow(<Header margin={false} />);
        expect(wrapper).toMatchSnapshot();
    });

    it('it renders with addition', () => {
        const wrapper = shallow(<Header addition={<div />} />);
        expect(wrapper).toMatchSnapshot();
    });

    it('it renders with children', () => {
        const wrapper = shallow(<Header><section /></Header>);
        expect(wrapper).toMatchSnapshot();
    });
});
