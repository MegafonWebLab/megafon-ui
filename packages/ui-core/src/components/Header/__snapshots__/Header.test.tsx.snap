// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> it renders with addition 1`] = `
<h1
  className="mfui-header mfui-header_color_black mfui-header_level_h1"
>
  <div
    className="mfui-header__addition"
  >
    <div />
  </div>
</h1>
`;

exports[`<Header /> it renders with children 1`] = `
<h1
  className="mfui-header mfui-header_color_black mfui-header_level_h1"
>
  <section />
</h1>
`;

exports[`<Header /> it renders with white color class 1`] = `
<h1
  className="mfui-header mfui-header_color_white mfui-header_level_h1"
/>
`;

exports[`<Header /> it renders without class margin 1`] = `
<h1
  className="mfui-header mfui-header_color_black mfui-header_level_h1"
/>
`;
